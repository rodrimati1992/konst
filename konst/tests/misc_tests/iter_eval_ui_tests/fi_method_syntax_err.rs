const R: core::ops::Range<usize> = 0..10;
const S: &[u8] = &[3, 5];

const _: () = {
    konst::iter::eval!(R, filter(), next());
    konst::iter::eval!(R, filter(||false), next());
    konst::iter::eval!(R, filter(|_|), next());
    konst::iter::eval!(R, filter(|_| false, 100), next());
    konst::iter::eval!(R, filter(|a, b|false), next());
};

const _: () = {
    konst::iter::eval!(R, find());
    konst::iter::eval!(R, find(||false));
    konst::iter::eval!(R, find(|_| false, 100));
    konst::iter::eval!(R, find(|a, b|false));
};

const _: () = {
    konst::iter::eval!(R, filter_map(), next());
    konst::iter::eval!(R, filter_map(||None), next());
    konst::iter::eval!(R, filter_map(|_|), next());
    konst::iter::eval!(R, filter_map(|_| None, 100), next());
    konst::iter::eval!(R, filter_map(|a, b|None), next());
};

const _: () = {
    konst::iter::eval!(R, find_map());
    konst::iter::eval!(R, find_map(||None));
    konst::iter::eval!(R, find_map(|_| None, 100));
    konst::iter::eval!(R, find_map(|a, b|None));
};


fn main(){}