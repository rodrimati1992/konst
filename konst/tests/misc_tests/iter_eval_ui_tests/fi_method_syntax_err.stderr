error: `filter` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:11:5
   |
11 |     konst::iter::eval!(R, filter(), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `filter` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:12:5
   |
12 |     konst::iter::eval!(R, filter(||false), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `filter` expects valid closure syntax, passed closure without return value
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:13:5
   |
13 |     konst::iter::eval!(R, filter(|_|), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_no_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `filter` expects no arguments after closure argument
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:14:5
   |
14 |     konst::iter::eval!(R, filter(|_| false, 100), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_trailing_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `filter` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:15:5
   |
15 |     konst::iter::eval!(R, filter(|a, b|false), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `filter` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:16:5
   |
16 |     konst::iter::eval!(R, filter({whatever1_return}), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `find` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:20:5
   |
20 |     konst::iter::eval!(R, find());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `find` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:21:5
   |
21 |     konst::iter::eval!(R, find(||false));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `find` expects no arguments after closure argument
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:22:5
   |
22 |     konst::iter::eval!(R, find(|_| false, 100));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_trailing_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `find` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:23:5
   |
23 |     konst::iter::eval!(R, find(|a, b|false));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `find` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:24:5
   |
24 |     konst::iter::eval!(R, find({whatever1_return}));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `filter_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:28:5
   |
28 |     konst::iter::eval!(R, filter_map(), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `filter_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:29:5
   |
29 |     konst::iter::eval!(R, filter_map(||None), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `filter_map` expects valid closure syntax, passed closure without return value
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:30:5
   |
30 |     konst::iter::eval!(R, filter_map(|_|), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_no_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `filter_map` expects no arguments after closure argument
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:31:5
   |
31 |     konst::iter::eval!(R, filter_map(|_| None, 100), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_trailing_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `filter_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:32:5
   |
32 |     konst::iter::eval!(R, filter_map(|a, b|None), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `filter_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:33:5
   |
33 |     konst::iter::eval!(R, filter_map({whatever1_return::<Option<u8>>}), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `find_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:37:5
   |
37 |     konst::iter::eval!(R, find_map());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `find_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:38:5
   |
38 |     konst::iter::eval!(R, find_map(||None));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `find_map` expects no arguments after closure argument
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:39:5
   |
39 |     konst::iter::eval!(R, find_map(|_| None, 100));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_trailing_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `find_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:40:5
   |
40 |     konst::iter::eval!(R, find_map(|a, b|None));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `find_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fi_method_syntax_err.rs:41:5
   |
41 |     konst::iter::eval!(R, find_map({whatever1_return::<Option<u8>>}));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)
